### PR TITLE
README: Update with figure descriptions

### PR DESCRIPTION
**Pull request by rleigh-codelibre**
I've added figure/result descriptions and a small number of conclusions.  May need further editing.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -154,20 +153,98 @@ for the assessment of each benchmark test:
...
-Additionally, we have reproduced the benchmark by repeating the same number of
-iterations of each test in a loop within the same environment. For most tests,
-the results were found to be identical and independent of whether the tests
-were run separately or repeated. Interestingly, in the case of the metadata
-tests using Bio-Formats and the Java Virtual Machine (JVM), there is a gain
-due to the optimizer in the JVM. We include these results for completeness and
-to indicate the performance achieved if the same operation is invoked within a
-process multiple times.
+The [analysis](analysis) folder contains the plots and summarized data
+after processing the data with GNU R and the script
```
Comment (sbesson) [inline README.md]: worth an extra link to https://www.r-project.org/

File: README.md [hunk 2]
```diff
@@ -154,20 +153,98 @@ for the assessment of each benchmark test:
...
+[plots.R](scripts/plots.R).
+
+[Figure 2](analysis/files-fig2.pdf) is a copy of the figure submitted
+for publication.  The read and write performance of pixeldata and
```
Comment (sbesson) [inline README.md]: double space

File: README.md [hunk 3]
```diff
@@ -154,20 +153,98 @@ for the assessment of each benchmark test:
...
+metadata are compared for three software implementations and three
+datasets.  The relative performance of Linux implementations are
```
Comment (sbesson) [inline README.md]: double space

File: README.md [hunk 4]
```diff
@@ -154,20 +153,98 @@ for the assessment of each benchmark test:
...
+being slower than Java in all cases, while C++ retained a slight edge
+over Java when writing.  JNI performance was faster than Java for the
+simplest dataset, and marginally faster for datasets with more complex
+metadata.
+
+For most tests, the results were found to be identical and independent
+of whether the tests were run separately or repeated. Interestingly,
+in the case of the metadata tests using Bio-Formats and the Java
+Virtual Machine (JVM), there is a gain due to the optimizer in the
+JVM.  This gain is small for pixeldata, but large for metadata.
```
Comment (sbesson) [inline README.md]: With the extended discussion on the results, this paragraph feels a bit out of place. I could see it either integrated in the previous paragraph (when discussing the Java observations) and/or as apart of the corresponding bullet point in the conclusions.

File: README.md [hunk 5]
```diff
@@ -154,20 +153,98 @@ for the assessment of each benchmark test:
...
+
+[Supplementary Figure 5](analysis/files-suppfig5.pdf) and
+[Supplementary Figure 6](analysis/files-suppfig6.pdf) present the same
+data in more detail as boxplots, showing the distribution of timings
+within each treatment.  Figure 6 shows the effects of the Java HotSpot
+optimizer when the same routines are invoked repeatedly.  The outliers
```
Comment (sbesson) [inline README.md]: Do we have an external reference for the HotSpot optimizer?

File: README.md [hunk 6]
```diff
@@ -154,20 +153,98 @@ for the assessment of each benchmark test:
...
+Summary data is presented in these tables:
+
+- [pixeldata](analysis/summary-pixeldata-repeated.tsv)
+- [metadata](analysis/summary-metadata-repeated.tsv)
+
+## Benchmark conclusions
+
+- C++ pixeldata reading and writing is significantly faster than Java
+  in all cases, indicating that for workloads requiring the fastest
+  possible data transfer rates that C++ should be preferred.
```
Comment (sbesson) [inline README.md]: extra `that` in this sentence

File: README.md [hunk 7]
```diff
@@ -154,20 +153,98 @@ for the assessment of each benchmark test:
...
+
+- C++ metadata reading showed poor scalability compared with Java.
+  Further profiling showed this was largely time spent in the
+  Xerces-C++ XML library, which appearts to have some inefficiencies
```
Comment (sbesson) [inline README.md]: `appears`

**Review by sbesson — COMMENTED** [review]
Overall, the references and discussions over the various tables and figures derived from the benchmark analysis scripts are very useful improvements. A few comments/proposals to improve the readability.

**Comment by rleigh-codelibre** [comment]
@sbesson The points you mentioned should all be fixed now.  I've additionally reflowed the text to make it readable outside the github markdown viewer in a terminal.

**Comment by sbesson** [comment]
All looks good. :+1: for the reflowing. Merging, tagging and opening a PR against master.

**Pull request merged**